components/org.wso2.carbon.identity.workflow.impl/src/main/java/org/wso2/carbon/identity/workflow/impl/RequestExecutor.java

```diff
@@ -108,6 +150,97 @@ private void validateExecutionParams() throws InternalWorkflowException {
             throw new InternalWorkflowException("Init params for the BPELExecutor is null.");
         }
     }
+    private String getExternalWorkflowId(String workflowId) throws WorkflowException {
+
+        return  WorkflowImplServiceDataHolder.getInstance().getWorkflowManagementService().
+                getExternalWorkflowId(workflowId);
+
+    }
+    private String getTemplateIdByWorkflowId(String workflowId) throws WorkflowException {
+        return  WorkflowImplServiceDataHolder.getInstance().getWorkflowManagementService().
+                getTemplateId(workflowId);
+    }
+    private WorkFlowRequest createWorkflowRequest(OMElement messagePayload) throws WorkflowException {
+
+
+        WorkFlowRequest workFlowRequest = new WorkFlowRequest();
+        workFlowRequest.setWorkflowID(getExternalWorkflowId(this.parameterList.get(0).getWorkflowId()));
+        List<Variable> parameterArray = new ArrayList<>();
+
+        Iterator workflowDetails = messagePayload.getChildElements();
+
+        while (workflowDetails.hasNext()) {
+            OMElementImpl workflowDetail = (OMElementImpl) workflowDetails.next();
+            if (PROCESS_UUID.equals(workflowDetail.getLocalName())) {
```

Comment:
We can improve the code to use switch case statements instead of multiple if conditions.

Reply:
addressed the comment ref:https://github.com/wso2-extensions/identity-workflow-impl-bps/pull/70/commits/641aaf120ea3a609b753a77ae028dae66a9ffa98